plutus-core/plutus-core/src/PlutusCore/Flat.hs

```diff
@@ -85,6 +88,15 @@ implementations for them (if they have any constructors reserved for future use)
 By default, Flat does not use any space to serialise `()`.
 -}
 
+newtype AsSerialize a = AsSerialize
+    { unAsSerialize :: a
```

Comment:
I've no idea what I'm doing, I only implemented @michaelpj's suggestion.

Reply:
- `Data` has a CBOR instance, because it's going to appear in a number of other places in transactions etc, and also it's literally designed to be a subset of CBOR. People may want to generate it using other tools etc.
- We need all types in the builtin universe to have `Flat` instances, so `Data` also needs a `Flat` instance. As Kenneth says, indeed this is so that we could serialize a program with a `Data` literal in it. For example, the fully-applied scripts produced by the `plutus-use-cases-scripts` thing will be like that.
- We *could* give `Data` a new `Flat` instance, but given that we already have a way of serializing it, we might as well just dump it in as an opaque serialized bytestring. That has the advantage that you could in principle fish it out and work with it separately, or whatever.